Milestone 5: Add friendships feature v1

descholar-ceo opened the pull request:
### What does this Pull Request do?
It adds the friendship feature to the stay in touch project
### Tasks to be accomplished on this Pull Request
- [x] Generate friendship migration, model and controller
- [x] Establish all associations between user table and itself through the friendship junction table
- [x] Establish controller actions which will help in friendship functionality
- [x] Write all necessary helper methods to help in friendship actions such as: `confirm friend request`, `reject friend request`, `retrieve a given user friends`...
- [x] Develop User Interface which facilitates the friendship functionality
- [x] Add unit tests and integration tests
- [x] Polish codes and fix any linting errors

addod19 commented on app/models/friendship.rb:
```diff
@@ -0,0 +1,4 @@
+class Friendship < ApplicationRecord
+  belongs_to :user, class_name: 'User'
```
Kindly remove the `class_name: 'User'` because rails will automatically identify it with the `:user` relationship

descholar-ceo replied:
Thank you so much for this help @addod19, we have just fixed this

addod19 reviewed (changes requested):
## Status: Require changes :recycle: :traffic_light: 

Hi, @descholar-ceo and @Georjane, amazing job with this project :+1: PR is very professional with a title and a short description :100: The project works as expected :fireworks: All models and associations have been set up correctly :+1: The code is well organized and clean. However, we need to address the comments below so we can get our project approved

- `Friendship model` we do not need `class_name: 'User'`. Kindly remove it
- `README` let us fix the typo in the test section. It reads `rpsec --format documentation`, kindly change it to `rspec --format documentation`
- Also, our test couldn't run successfully. It outputs the following error `cannot load such file -- rspec/core/formatters/profile_formatter (LoadError)`. Let us kindly fix it
I was able to run the test with the following command `bundle exec rspec`

Kindly submit another code review once this is resolved :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

descholar-ceo commented:
> ## Status: Require changes  
> Hi, @descholar-ceo and @Georjane, amazing job with this project  PR is very professional with a title and a short description  The project works as expected  All models and associations have been set up correctly  The code is well organized and clean. However, we need to address the comments below so we can get our project approved
> 
> * `Friendship model` we do not need `class_name: 'User'`. Kindly remove it
> * `README` let us fix the typo in the test section. It reads `rpsec --format documentation`, kindly change it to `rspec --format documentation`
> * Also, our test couldn't run successfully. It outputs the following error `cannot load such file -- rspec/core/formatters/profile_formatter (LoadError)`. Let us kindly fix it
>   I was able to run the test with the following command `bundle exec rspec`
> 
> Kindly submit another code review once this is resolved 
> _Happy coding_ 
> [Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

@addod19 Thank you so much for this kind feedback, it got us to update and made our work even better, we implemented the requested changes, and added the right command of running tests which will be `bundle exec rspec --format documentation` which seems to work everywhere.

rOluochKe reviewed (approved):
Hello Team
### You got this right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!
